src/evict/evict_lru.c

```diff
@@ -1915,13 +1915,25 @@ __evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_ent
         if (__wt_page_is_empty(page) || F_ISSET(session->dhandle, WT_DHANDLE_DEAD))
             goto fast;
 
-        /* Skip clean pages if appropriate. */
-        if (!modified && !F_ISSET(cache, WT_CACHE_EVICT_CLEAN))
-            continue;
+        {
+            // TODO fix this so it reads cleanly.
```

Comment:
This needs a tidy - but the structure is easy to follow, I don't think you need to do much.